plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt

```diff
@@ -415,37 +411,93 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
     centerScreen = ScreenCoordinate((width / 2).toDouble(), (height / 2).toDouble())
   }
 
+  /**
+   * Used instead of [com.mapbox.android.gestures.StandardGestureDetector.SimpleStandardOnGestureListener]
+   * since it adds NonNull annotations while its parent [android.view.GestureDetector.OnGestureListener] does not
+   * have those annotations in the release Android versions (added recently at https://android.googlesource.com/platform/frameworks/base/+/68c65e58b4f4daf79c9ffab518a826a506799db2/core/java/android/view/GestureDetector.java).
+    */
+  private open class SimpleStandardOnGestureListener : StandardOnGestureListener {
+    override fun onSingleTapConfirmed(motionEvent: MotionEvent?): Boolean {
+      return false
+    }
+
+    override fun onDoubleTap(motionEvent: MotionEvent?): Boolean {
+      return false
+    }
+
+    override fun onDoubleTapEvent(motionEvent: MotionEvent?): Boolean {
+      return false
+    }
+
+    override fun onDown(motionEvent: MotionEvent?): Boolean {
+      return false
+    }
+
+    override fun onShowPress(motionEvent: MotionEvent?) {}
+
+    override fun onSingleTapUp(motionEvent: MotionEvent?): Boolean {
+      return false
+    }
+
+    override fun onScroll(
+      e1: MotionEvent?,
+      e2: MotionEvent?,
+      distanceX: Float,
+      distanceY: Float
+    ): Boolean {
+      return false
+    }
+
+    override fun onLongPress(motionEvent: MotionEvent?) {}
+
+    override fun onFling(
+      e1: MotionEvent?,
+      e2: MotionEvent?,
+      velocityX: Float,
+      velocityY: Float
+    ): Boolean {
+      return false
+    }
+  }
+
   /**
    * Standard gesture listener, receives callbacks for gestures detected by AndroidGesturesManager.
    */
-  private inner class StandardGestureListener internal constructor(private val doubleTapMovementThreshold: Float) :
-    StandardGestureDetector.SimpleStandardOnGestureListener() {
+  private inner class StandardGestureListener(private val doubleTapMovementThreshold: Float) :
+    SimpleStandardOnGestureListener() {
 
     /**
      * Called when an on down gesture was detected.
      */
-    override fun onDown(motionEvent: MotionEvent): Boolean {
+    override fun onDown(motionEvent: MotionEvent?): Boolean {
```

Comment:
why don't return `true` in superclass and not even overwrite it here then?

Reply:
Dont want to mix logic here. 
Logic is added in the `StandardGestureListener`, while `SimpleStandardOnGestureListener` is just a class implementing everything and returning `false`. If/when we'll fix in in the gestures lib (which seams not feasible right now since will be non-backward compatible) we'll get rid of out Simple listener

Reply:
> which seams not feasible right now since will be non-backward compatible)

I think if this is a bug, e.g the MotionEvent in Android GestureDetector is not marked as NonNull
```
    public interface OnGestureListener {

        /**
         * Notified when a tap occurs with the down {@link MotionEvent}
         * that triggered it. This will be triggered immediately for
         * every down event. All other events should be preceded by this.
         *
         * @param e The down motion event.
         */
        boolean onDown(MotionEvent e);
```
and it is marked as NonNull in the SimpleStandardOnGestureListener in the mapbox-gestures library
```
    @Override
    public boolean onSingleTapConfirmed(@NonNull MotionEvent e) {
      return false;
    }
```
We can probably just remove the NonNull annotation and consider it a bug fix, instead of a API breaking change. wdyt?

Reply:
Well, still a breaking change, not sure we need to introduce it if its possible to fix on our side. wdyt? cc @mapbox/maps-android

Reply:
I would prefer we fix the bug on our side and add the issue link of gesture lib in the comment.

Reply:
> still a breaking change

Maybe we can release a major version of the gestures library? but yeah, we have gestures library as an api dependency, so these APIs are also part of our public APIs.

Imo if something is clearly broken without API changes, it would be dangerous to leave it untouched, user still will be able to find this class and use it. Maybe @LukasPaczos has some suggestions/recommendations as well?

Reply:
I agree with @pengdev that this should be fixed in the gestures library. The current behavior is incorrect and cannot be justified as a use-case that we'd be breaking.

The gestures library is still at v0.x (probably shouldn't at this point :upside_down_face:) so to me the best solution is a v0.8.0 with fixed annotations and a clear communication in both gestures lib and Maps SDK changelogs that the API compatibility can be impacted for someone that overrides the listeners.

Reply:
Looks like the annotations themselves are a regression from https://github.com/mapbox/mapbox-gestures-android/pull/95, we shouldn't have inferred the nullity for non-annotated Java APIs from the Android SDK.

Reply:
Okay, then lets move this fix to the gestures instead.

Reply:
Moved fix there https://github.com/mapbox/mapbox-gestures-android/pull/101